apps/tasks/lib/screens/room_overview_screen.dart

```diff
@@ -1,47 +1,243 @@
-import 'dart:math';
-
 import 'package:flutter/material.dart';
 import 'package:tasks/components/navigation_drawer.dart';
 import 'package:tasks/components/bed_card.dart';
+import 'package:helpwave_widget/shapes.dart';
+import 'package:helpwave_localization/localization.dart';
 
-/// Screen for showing all [Task]'s within a [Ward] grouped by [Room]'s
-class RoomOverviewScreen extends StatelessWidget {
-  final Random _random = Random();
+// TODO replace all these DTOs with real grpc generated onces
+class RoomDTO {
+  final String name;
+  final List<PatientDTO> patients;
 
-  RoomOverviewScreen({super.key});
+  RoomDTO({
+    required this.name,
+    required this.patients,
+  });
+}
+
+class WardDTO {
+  final String name;
+  final List<RoomDTO> rooms;
 
-  int _getRandomNumber({int min = 0, int max = 20}) =>
-      min + _random.nextInt(max - min);
+  WardDTO({
+    required this.name,
+    required this.rooms,
+  });
+}
+
+/// A widget containing a room title bar and the associated bed cards
+class _RoomSection extends StatelessWidget {
+  final RoomDTO room;
+
+  const _RoomSection({
+    required this.room,
+  });
 
   @override
   Widget build(BuildContext context) {
     double width = MediaQuery.of(context).size.width * 0.5;
-    double height = 100;
+    double height = 85;
     double ratio = width / height;
 
-    return Scaffold(
-      drawer: const TasksNavigationDrawer(
-        currentPage: NavigationOptions.roomoverview,
+    return ListView(
+      shrinkWrap: true,
+      physics: const NeverScrollableScrollPhysics(),
+      children: [
+        Container(height: 10),
+        Row(
+          children: [
+            const Spacer(flex: 1),
+            const Circle(
+              color: Colors.blue, // TODO replace with primary color
+              diameter: 20,
+            ),
+            const Spacer(flex: 1),
+            Text(
+              "${context.localization!.room} ${room.name}",
+              style: const TextStyle(
+                // TODO use SpaceGrotesk here
+                fontWeight: FontWeight.bold,
+                fontSize: 15,
+              ),
+            ),
+            const Spacer(flex: 15),
+          ],
+        ),
+        room.patients.isNotEmpty
+            ? GridView.count(
+                childAspectRatio: ratio,
+                shrinkWrap: true,
+                physics: const NeverScrollableScrollPhysics(),
+                padding: const EdgeInsets.all(10),
+                crossAxisCount: 2,
+                children: room.patients
+                    .map((PatientDTO patient) => BedCard(patient: patient))
+                    .toList(),
+              )
+            : Padding(
+                padding: const EdgeInsets.only(left: 20, top: 10, bottom: 10),
+                child: Text(
+                  context.localization!.noBedAvailable,
+                  style: const TextStyle(
+                    fontWeight: FontWeight.w300,
+                  ),
```

Comment:
dont use custom fontWeight use a TextStyle from texttheme in stead or mark with a todo for the texttheme/themeUpdate